app/models/team/Team.scala

```diff
@@ -138,15 +138,20 @@ object TeamSQLDAO extends SQLDAO[TeamSQL, TeamsRow, Teams] {
 case class Team(name: String,
                 organization: String,
                 isOrganizationTeam: Boolean = false,
-                _id: BSONObjectID = BSONObjectID.generate) {
+                _id: BSONObjectID = BSONObjectID.generate) extends FoxImplicits {
 
   lazy val id = _id.stringify
 
-  def couldBeAdministratedBy(user: User) =
-    user.organization == organization
+  def couldBeAdministratedBy(user: UserSQL) =
+    for {
+      organization <- user.organization
+    } yield organization.name == this.organization
 
-  def isAdminOfOrganization(user: User) =
-    user.organization == organization && user.isAdmin
+  def assertCouldBeAdministratedBy(user: UserSQL) =
+    for {
+      asBoolean <- couldBeAdministratedBy(user)
```

Comment:
This syntax is not very intuitive to me. Having something like the following would be awesome (in my opinion, of course):
```
async def assertCouldBeAdministratedBy(user: UserSQL) = 
  if (await couldBeAdministratedBy(user))
    Success()
  else {
    // not sure with the syntax here, but something like
    throw Messages("notAllowed")
    Failure()
  }
```

a) What is your opinion about this, @fm3 ?
b) Is something similar possible?

Reply:
(having the bool2fox conversion explicitely would help to make it clearer, but I still find an `if` statement appropriate)

Reply:
should be `Fox.assertTrue(team.couldBeAdministratedBy(user))`. will change.

Reply:
:+1: